src/Ocelot/Values/ServiceHostAndPort.cs

```diff
@@ -1,20 +1,54 @@
-﻿namespace Ocelot.Values
+﻿namespace Ocelot.Values;
+
+public class ServiceHostAndPort : IEquatable<ServiceHostAndPort>
 {
-    public class ServiceHostAndPort
+    public ServiceHostAndPort(ServiceHostAndPort from)
     {
-        public ServiceHostAndPort(string downstreamHost, int downstreamPort)
-        {
-            DownstreamHost = downstreamHost?.Trim('/');
-            DownstreamPort = downstreamPort;
-        }
+        DownstreamHost = from.DownstreamHost;
+        DownstreamPort = from.DownstreamPort;
+        Scheme = from.Scheme;
+    }
 
-        public ServiceHostAndPort(string downstreamHost, int downstreamPort, string scheme)
-            : this(downstreamHost, downstreamPort) => Scheme = scheme;
+    public ServiceHostAndPort(string downstreamHost, int downstreamPort)
+    {
+        DownstreamHost = downstreamHost?.Trim('/');
+        DownstreamPort = downstreamPort;
+    }
 
-        public string DownstreamHost { get; }
+    public ServiceHostAndPort(string downstreamHost, int downstreamPort, string scheme)
+        : this(downstreamHost, downstreamPort) => Scheme = scheme;
 
-        public int DownstreamPort { get; }
-
-        public string Scheme { get; }
-    }
-}
+    public string DownstreamHost { get; }
+    public int DownstreamPort { get; }
+    public string Scheme { get; }
+
+    public override string ToString()
+        => $"{Scheme}:{DownstreamHost}:{DownstreamPort}";
+    public override int GetHashCode()
+        => Tuple.Create(Scheme, DownstreamHost, DownstreamPort).GetHashCode();
+
+    public bool Equals(ServiceHostAndPort other) => this == other;
+    public override bool Equals(object obj)
+        => obj != null && obj is ServiceHostAndPort o && this == o;
+
+    /// <summary>Checks equality of two hosts.</summary>
+    /// <remarks>Microsoft Learn | .NET | C# Docs:
+    ///   <list type="bullet">
+    ///   <item><seealso href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/equality-operators">Equality operators</seealso></item>
+    ///   <item><seealso href="https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-object-equals">System.Object.Equals method</seealso></item>
+    ///   <item><seealso href="https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1.equals?view=net-8.0">IEquatable&lt;T&gt;.Equals(T) Method</seealso></item>
+    ///   </list>
+    /// </remarks>
+    /// <param name="l">Left operand.</param>
+    /// <param name="r">Right operand.</param>
+    /// <returns><see langword="true"/> if both operands are equal; otherwise, <see langword="false"/>.</returns>
+    public static bool operator ==(ServiceHostAndPort l, ServiceHostAndPort r)
+        => (((object)l) == null || ((object)r) == null)
+            ? object.Equals(l, r)

```

Comment:
```object``` is redundant

Reply:
**P.S.** Probably you mean: `object.Equals(l, r)` &rarr; `Equals(l, r)` ? Then 🆗I will fix this.

Reply:
@raman-m yes 😊

Reply:
Already fixed in https://github.com/ThreeMammals/Ocelot/pull/2111/commits/c4c7ae0ff2a891d64b4f81b6a6d051cc15e71211#diff-5baf30f066ec9efc9275e78bd80143b4c967bda577cbff011f1580598f8f80e5